docs/api.rst

```diff
@@ -6,7 +6,38 @@
 
 
 LIBBPF API
-==================
+==========
+
+Error Handling
+--------------
+
+API functions in libbpf can return errors in one of two ways. 
+
+If the function returns an error code directly, it uses 0 to indicate success
+and a negative error code to indicate what caused the error. In this case the
+error code should be checked directly from the return, you do not need to check
+errno. 
+
+For example:
+
+```
+    err = some_libbpf_api_with_error_return(...);
+    if (err < 0) {
+        // Handle error accordingly
+    }
+```
+
+If the function returns a pointer, it will return NULL to indicate there was 
+an error. In this case errno should be checked for the error code.
+
+For example:
+
+```
+    ptr = some_libbpf_api_returning_ptr(...);
+    if (!ptr) {
+       /* errno has error */ 
```

Comment:
I guess add a code like `if (errno == EINVAL) { /* handle error */ }` to make it unambiguous what "errno" we refer to?